### PR TITLE
Add docs for java+database example

### PR DESCRIPTION
**Pull request by dharmit**
**What type of PR is this?**
/kind documentation

**What does this PR do / why we need it**:
Updates the odo.dev website to use latest updates to the documentation 

**Which issue(s) this PR fixes**:

Fixes NA

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [x] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: docs/deploying-java-app-with-database.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    
+    You can inspect the new intermediate secret via the dashboard console in the 'my-postgresql-operator-dev4devs-com' namespace by navigating to Secrets and clicking on the secret named `mysboproj-database-sampledatabase` Notice it contains 4 pieces of data all related to the connection information for your PostgreSQL database instance.
+
+5.  Push the newly created link. This will terminate the existing application pod and start a new application pod.
+    
+    ``` sh
+    odo push
+    ```
+
+6.  Once the new pod has initialized you can see the secret database connection data as it is injected into the pod environment by executing the following:
```
Comment (valaparthvi) [inline docs/deploying-java-app-with-database.md:245]: I think adding a comma after "Once the new pod has initialized" will make more sense.

File: docs/deploying-java-app-with-database.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    
+    ``` sh
+    $ odo exec -- bash -c 'export | grep DATABASE'
+    declare -x DATABASE_CLUSTERIP="10.106.182.173"
+    declare -x DATABASE_DB_NAME="sampledb"
+    declare -x DATABASE_DB_PASSWORD="samplepwd"
+    declare -x DATABASE_DB_USER="sampleuser"
+    ```
+    
+    Once the new version is up (there will be a slight delay until application is available), navigate to the CreatePerson.xhtml using the URL created in a previous step. Enter requested data and click the **Save** button.
```
Comment (valaparthvi) [inline docs/deploying-java-app-with-database.md:255]: "(there will be a slight delay until **the** application is available)" ?

File: docs/deploying-java-app-with-database.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    
+    Notice you are re-directed to the `PersonList.xhtml` page, where your data is displayed having been input to the postgreSQL database and retrieved for display purposes.
```
Comment (valaparthvi) [inline docs/deploying-java-app-with-database.md:257]: "Notice **that** you .."

File: docs/deploying-java-app-with-database.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+    
+    You may inspect the database instance itself and query the table to see the data in place by using the postgreSQL command line tool, psql.
```
Comment (valaparthvi) [inline docs/deploying-java-app-with-database.md:259]: Perhaps highlight the "psql" part?

File: docs/deploying-java-app-with-database.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+7.  Navigate to the pod containing your db from the Kubernetes Dashboard
```
Comment (valaparthvi) [inline docs/deploying-java-app-with-database.md:261]: Missing full-stop at the end.

File: docs/deploying-java-app-with-database.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+
+8.  Click on the terminal tab.
+
+9.  At the terminal prompt access psql for your database
```
Comment (valaparthvi) [inline docs/deploying-java-app-with-database.md:265]: "9.  At the terminal prompt, access psql for your database."
Missing comma before "access" and a full stop or color at the end.

File: docs/installing-and-onfiguring-minikube-environment.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+# Hero section
+title: Setup the minikube environment
+description: Setup a Kubernetes cluster that odo can be used with
+
+# Micro navigation
+micro_nav: true
+---
+> **Note**
+> 
+> The guide only works with minikube configured with kubernetes 1.19.x or lower. `odo link` cannot link services successfully in a Kubernetes 1.20.x or newer environment.
```
Comment (valaparthvi) [inline docs/installing-and-onfiguring-minikube-environment.md:16]: >kubernetes 1.19.x or lower

Kubernetes with a **K** looks better than with a **k** IMO.

File: docs/installing-and-onfiguring-minikube-environment.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
+---
```
Comment (valaparthvi) [inline docs/installing-and-onfiguring-minikube-environment.md:1]: On another not-so-related note, this file too could use an update in the name. I guess it should be "installing-and-**c**onfiguring-minikube-environment.md".

File: docs/installing-and-onfiguring-minikube-environment.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+    $ minikube start --kubernetes-version=v1.19.8
+    ```
+
+# Configuring minikube
+
+## Enabling ingress addon
+
+The application requires an ingress addon to allow the routes to be created easily. It enables `odo url` commands.
+
+1.  onfigure minikube for ingress by adding ingress as a minikube add-on:
```
Comment (valaparthvi) [inline docs/installing-and-onfiguring-minikube-environment.md:44]: > 1.  onfigure

I think the "C" key on your keyboard is dead :P

File: docs/installing-and-onfiguring-minikube-environment.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+## Adding a pull secret to ingress accounts
+
+You may face the DockerHub pull rate limit if you do not have a pull secret for your personal free DockerHub account. During ingress initialization two of the job pods used by ingress may fail to initialize due to pull rate limits. If this happens, and ingress fails to enable, you add a secret for the pulls for the following service accounts:
+
+  - ingress-nginx-admission
+
+  - ingress-nginx
+
+to add a pull secret for these service accounts:
```
Comment (valaparthvi) [inline docs/installing-and-onfiguring-minikube-environment.md:58]: "**To** add a..."?

File: docs/installing-and-onfiguring-minikube-environment.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+2.  Create a pull secret:
+    
+    ``` sh
+    $ kubectl create secret docker-registry regcred --docker-server=<your-registry-server> --docker-username=<your-name> --docker-password=<your-pword> --docker-email=<your-email>
+    ```
+    
+    where:
+    
+      - \<your-registry-server\> is the DockerHub Registry FQDN. (<https://index.docker.io/v1/>)
```
Comment (valaparthvi) [inline docs/installing-and-onfiguring-minikube-environment.md:74]: https://index.docker.io/v1/ is a dead-end. Is it correct or just an example?

File: docs/installing-and-onfiguring-minikube-environment.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+
+  - Create the same docker-registry secret configured for your docker, now for the default minikube context:
+    
+    ``` sh
+    $ kubectl create secret docker-registry regcred --docker-server=<your-registry-server> --docker-username=<your-name> --docker-password=<your-pword> --docker-email=<your-email>
+    ```
+    
+    where:
+    
+      - \<your-registry-server\> is the DockerHub Registry FQDN. (<https://index.docker.io/v1/>)
```
Comment (valaparthvi) [inline docs/installing-and-onfiguring-minikube-environment.md:107]: Same as above.

**Review by valaparthvi — CHANGES_REQUESTED** [review]
Great work on the documentation!
And I am sorry about all the petty comments regarding missing commas, full-stops, and the-s on the PR!

**Comment by dharmit** [comment]
@valaparthvi this PR is raised for `gh-pages` branch. The feedback you've shared would be applicable for a docs PR on `master` branch.

The way it works is:
1. someone opens a docs PR on `master` branch
1. we review the PR (just like you did here)
1. then using a script, we convert the docs to be usable on odo.dev website 
1. open a PR on `gh-pages` branch (this PR is mostly just namesake. We don't review it in the usual sense and in the past, Charlie often used to forcefully merge it to expedite pushing things)

The last two steps will be automated someday by https://github.com/openshift/odo/issues/4255.

So in this case, the actual PR where your feedback would have been really helpful is #4486.

Since you've made efforts on this one, it would make sense to open a fresh PR against `master` branch and follow the steps as I listed above.

**Comment by valaparthvi** [comment]
Ack. I think I'll raise a PR this week with the changes that I suggested here. It LGTM otherwise.

/lgtm

**Pull request merged**